### PR TITLE
Remove a component and its statements from a system's selected components

### PR DESCRIPTION
**Pull request by gregelin**
Remove a component and its statements from a system's selected components
  
Add button to selected component pages to remove
a previously selected component from the system's
selected components.

Also, relocate the "Add a component" select2 drop down
widget to the top of the selected component page.

TODO: 
- Write tests

### CODE REVIEW

**Comment by gregelin** [comment]
@davidpofo Can you review this PR? And also, I need your help to update views.SelectedComponentsList.as_view to pass
value if user has permission to change page in order to optionally display the remove component trash can button.

File: controls/views.py [hunk 1]
```diff
@@ -700,6 +700,52 @@ def system_element(request, system_id, element_id):
...
+def system_element_remove(request, system_id, element_id):
+    """Remove an element from a system and delete the controls it produces"""
+
+    # Retrieve identified System
+    system = System.objects.get(id=system_id)
+    # Retrieve related selected controls if user has permission on system
+    if request.user.has_perm('change_system', system):
+        # Retrieve primary system Project
+        # Temporarily assume only one project and get first project
+        project = system.projects.all()[0]
```
Comment (davidpofo) [inline controls/views.py:713]:
```suggestion
```
not used

File: controls/views.py [hunk 2]
```diff
@@ -700,6 +700,52 @@ def system_element(request, system_id, element_id):
...
+
+        # Retrieve element
+        element = Element.objects.get(id=element_id)
+
+        # Retrieve impl_smts produced by element and consumed by system
+        # Get the impl_smts contributed by this component to system
+        impl_smts = element.statements_produced.filter(consumer_element=system.root_element)
+
+        # Delete the control implementation statements associated with this component
+        result = element.statements_produced.filter(consumer_element=system.root_element).delete()
```
Comment (davidpofo) [inline controls/views.py:723]:
```suggestion
        impl_smts = element.statements_produced.filter(consumer_element=system.root_element)

        # Delete the control implementation statements associated with this component
        result = impl_smts .delete()
```

File: controls/views.py [hunk 3]
```diff
@@ -700,6 +700,52 @@ def system_element(request, system_id, element_id):
...
+        logger.info(
+                event="change_system remove_component",
+                object={"object": "component", "id": element.id},
+                user={"id": request.user.id, "username": request.user.username}
+                )
+
+        # Create message for user
+        messages.add_message(request, messages.INFO, f"Removed component '{element.name}' from system.")
+
+        response = redirect(f'/controls/{system_id}/components/selected')
```
Comment (davidpofo) [inline controls/views.py:735]: use the url pattern name instead of hard-code

File: controls/views.py [hunk 4]
```diff
@@ -700,6 +700,52 @@ def system_element(request, system_id, element_id):
...
+        return response
+
+    else:
+        # User does not have permission
+
+        # Create message for user
+        messages.add_message(request, messages.INFO, f"You do not have permission to edit the system.")
+
```
Comment (davidpofo) [inline controls/views.py:743]: need to log lack of permissions

File: templates/systems/components_selected.html [hunk 5]
```diff
@@ -22,6 +22,26 @@
     <div class="systems-top">
             <div id="tab-content" class="row rows-header">
                 <div id="" class="col-xs-3 col-sm-3 col-md-3 col-lg-3 col-xl-3"><span class="systems-selected-components">Selected Components</span></div>
+                <div id="" class="col-xs-3 col-sm-3 col-md-3 col-lg-3 col-xl-3">&nbsp;</div>
+                <div id="" class="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-6 pull-right">
+                    <form id="add_component" class="form-inline" method="post" action="/systems/{{ system.id }}/components/add_system_component">
```
Comment (davidpofo) [inline templates/systems/components_selected.html:27]: url pattern instead of hard-code

File: templates/systems/components_selected.html [hunk 6]
```diff
@@ -30,34 +50,19 @@
...
                         </div>
-                        <div id="" class="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-6">
+                        <div id="" class="col-xs-5 col-sm-5 col-md-5 col-lg-5 col-xl-5">
                             {% if component.description %}{{ component.description }}{% else %}<span class="not-provided">No description provided.</span>{% endif %}
                         </div>
                         <div id="" class="col-xs-2 col-sm-2 col-md-2 col-lg-2 col-xl-2">
                             <span class="pull-right">{% if component.get_control_impl_smts_prototype_count > 0 %}{{ component.get_control_impl_smts_prototype_count }} control{{ component.get_control_impl_smts_prototype_count|pluralize }}</span>{% else %}None{% endif %}
                         </div>
+                        <div id="" class="col-xs-1 col-sm-1 col-md-1 col-lg-1 col-xl-1 pull-right">
+                            <a href="{% url 'system_element_remove' system_id=system.id element_id=component.id %}"><small><span class="glyphicon glyphicon-trash" title="remove component"></span></small></a>
```
Comment (davidpofo) [inline templates/systems/components_selected.html:60]: Kerry said just don't show it if the user doesn't have permission

**Review by davidpofo — COMMENTED** [review]
@gregelin PR party 2/25 almost there.

File: controls/views.py (hunk 2 above)
Comment (gregelin) [reply controls/views.py]: Fixed.

File: templates/systems/components_selected.html (hunk 5 above)
Comment (gregelin) [reply templates/systems/components_selected.html:27]: Fixed to use url reverse.

File: templates/systems/components_selected.html (hunk 6 above)
Comment (gregelin) [reply templates/systems/components_selected.html:60]: Waiting on @davidpofo to make update...

File: controls/views.py (hunk 4 above)
Comment (gregelin) [reply controls/views.py:741]: Done.

File: templates/systems/components_selected.html (hunk 6 above)
Comment (davidpofo) [reply templates/systems/components_selected.html:62]: done with django-guardian

**Comment by davidpofo** [comment]
lgtm

**Pull request merged**